src/views/catalog/customize/CustomizeVirtualMachine.tsx

```diff
@@ -0,0 +1,185 @@
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { useURLParams } from '@kubevirt-utils/hooks/useURLParams';
+import { useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
+import {
+  ActionGroup,
+  Button,
+  ExpandableSection,
+  Form,
+  FormGroup,
+  Sidebar,
+  SidebarContent,
+  SidebarPanel,
+  Stack,
+  StackItem,
+  TextInput,
+  Title,
+} from '@patternfly/react-core';
+import ExternalLinkAltIcon from '@patternfly/react-icons/dist/esm/icons/external-link-alt-icon';
+
+import { CustomizeVirtualMachineScheleton } from './components/CustomizeVirtualMachineScheleton';
+
+import '@patternfly/patternfly/patternfly.css';
+import './CustomizeVirtualMachine.scss';
+
+type RightHeadingProps = {
+  template: V1Template;
+};
+
+const RightHeader: React.FC<RightHeadingProps> = ({ template }) => {
+  const { t } = useKubevirtTranslation();
+  if (!template?.metadata?.annotations) return null;
+
+  const title = template.metadata?.annotations['openshift.io/display-name'];
+  const documentationLink = template.metadata?.annotations['openshift.io/documentation-url'];
+  const description = template.metadata?.annotations?.description;
+  return (
+    <Stack className="customize-vm__right-header">
+      <StackItem>
+        <Title headingLevel="h1">{title}</Title>
+      </StackItem>
+      <StackItem>
+        <Button
+          variant="link"
+          icon={<ExternalLinkAltIcon />}
+          href={documentationLink}
+          target="_blank"
+          component="a"
+          iconPosition="right"
+          className="pf-u-pl-0"
+        >
+          {t('View documentation')}
+        </Button>
+      </StackItem>
+
+      <StackItem>
+        <p>{description}</p>
+      </StackItem>
+    </Stack>
+  );
+};
+
+const FieldGroup: React.FC<TemplateParameter> = (field) => {
+  const { name, description, displayName, required, value: initialValue } = field;
+  const [value, setValue] = React.useState(initialValue);
+
+  return (
+    <FormGroup
+      label={displayName}
+      fieldId={`customize-required-${name}`}
+      isRequired={required}
+      helperText={description}
+    >
+      <TextInput
+        isRequired={required}
+        type="text"
+        id={name}
+        name={name}
+        aria-describedby="simple-form-name-01-helper"
+        value={value}
+        onChange={(newValue) => setValue(newValue)}
+      />
+    </FormGroup>
+  );
+};
+
+type CustomizeFormProps = {
+  template: V1Template;
+};
+
+const CustomizeForm: React.FC<CustomizeFormProps> = ({ template }) => {
+  const { t } = useKubevirtTranslation();
+  const [optionalFieldsExpanded, setOptionalFieldsExpanded] = React.useState(true);
+  const optionalFields = template.parameters?.filter((parameter) => !parameter.required);
+  const requiredFields = template.parameters?.filter((parameter) => parameter.required);
+
+  return (
+    <Form>
+      {requiredFields?.map((field) => (
+        <FieldGroup
```

Comment:
@yaacov @metalice @avivtur  we need to implement form validation infrastructure.
I'll try to think of the best way , let me know what you think

Reply:
Yes. in this case it's not complicated. I have only to be sure that the field is not empty. But in the future, we could have more complicated validations

Reply:
> But in the future, we could have more complicated validations

Validation in the UI should be expicitly `truthy`, so apart from checking for valid names (see documentaion: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/ ) we can't do much without sending to the backend.